zebra-network/src/config.rs

```diff
@@ -225,7 +225,7 @@ impl Config {
     pub fn initial_peer_hostnames(&self) -> &IndexSet<String> {
         match self.network {
             Network::Mainnet => &self.initial_mainnet_peers,
-            Network::Testnet => &self.initial_testnet_peers,
+            Network::Testnet(_) => &self.initial_testnet_peers,
```

Comment:
This is incorrect and could cause issues with the reliability of the public testnet. A custom testnet needs to supply its own peers if it changes certain parameters.

Currently those are the genesis hash, the activation heights, and the network magic. Let's write code that checks if these parameters match the default testnet parameters?

There's a specific way to do that so that each newly added parameter field causes a compile error, and we have to add it to the check.

Reply:
I think the specific check that we want here is:
If the network is an incompatible custom testnet (that is, it changes any incompatible parameters from the default testnet values), and the configured initial testnet peers contains any of the default testnet peers, then panic.